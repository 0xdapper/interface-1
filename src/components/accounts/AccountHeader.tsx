import { DrawerActions } from '@react-navigation/core'
import { selectionAsync } from 'expo-haptics'
import React, { useCallback } from 'react'
import { useAppDispatch, useAppSelector, useAppTheme } from 'src/app/hooks'
import { useEagerActivityNavigation } from 'src/app/navigation/hooks'
import { useAppStackNavigation } from 'src/app/navigation/types'
import ScanQRWCIcon from 'src/assets/icons/scan-qr-wc.svg'
import { AddressDisplay } from 'src/components/AddressDisplay'
import { TouchableArea } from 'src/components/buttons/TouchableArea'
import { Chevron } from 'src/components/icons/Chevron'
import { TxHistoryIconWithStatus } from 'src/components/icons/TxHistoryIconWithStatus'
import { Flex } from 'src/components/layout'
import { Box } from 'src/components/layout/Box'
import { ScannerModalState } from 'src/components/QRCodeScanner/constants'
import { FEATURE_FLAGS } from 'src/features/experiments/constants'
import { useFeatureFlag } from 'src/features/experiments/hooks'
import { openModal } from 'src/features/modals/modalSlice'
import { PendingNotificationBadge } from 'src/features/notifications/PendingNotificationBadge'
import { ElementName, ModalName } from 'src/features/telemetry/constants'
import { useSortedPendingTransactions } from 'src/features/transactions/hooks'
import { selectActiveAccountAddress } from 'src/features/wallet/selectors'
import { removePendingSession } from 'src/features/walletConnect/walletConnectSlice'
import { iconSizes } from 'src/styles/sizing'

function QRScannerIconButton({ onPress }: { onPress: () => void }) {
  const theme = useAppTheme()

  return (
    <TouchableArea hapticFeedback name={ElementName.WalletConnectScan} onPress={onPress}>
      <ScanQRWCIcon
        color={theme.colors.textSecondary}
        height={theme.iconSizes.lg}
        strokeWidth={2}
      />
    </TouchableArea>
  )
}

export function AccountHeader() {
  const theme = useAppTheme()
  const navigation = useAppStackNavigation()
  const activeAddress = useAppSelector(selectActiveAccountAddress)
  const dispatch = useAppDispatch()

  const { preload, navigate } = useEagerActivityNavigation()

  const sortedPendingTransactions = useSortedPendingTransactions(activeAddress)

  const accountSwitcherModalEnabled = useFeatureFlag(FEATURE_FLAGS.AccountSwitcherModal, false)
  const onPressAccountHeader = useCallback(() => {
    if (accountSwitcherModalEnabled) {
      dispatch(openModal({ name: ModalName.AccountSwitcher }))
    } else {
      navigation.dispatch(DrawerActions.openDrawer())
    }
  }, [accountSwitcherModalEnabled, dispatch, navigation])

  const onPressNotifications = useCallback(() => {
    if (activeAddress) {
      navigate()
    }
  }, [activeAddress, navigate])

  const onPressInNotifications = useCallback(() => {
    if (activeAddress) {
      preload(activeAddress)
    }
  }, [activeAddress, preload])

  const onPressScan = useCallback(() => {
    // in case we received a pending session from a previous scan after closing modal
    dispatch(removePendingSession())
    dispatch(
      openModal({ name: ModalName.WalletConnectScan, initialState: ScannerModalState.ScanQr })
    )
  }, [dispatch])

  return (
    <Box
      alignItems="center"
      flexDirection="row"
      justifyContent="space-between"
      mt="md"
      testID="account-header">
      <TouchableArea
        alignItems="center"
        flex={1}
        flexDirection="row"
        name={ElementName.Manage}
        testID={ElementName.Manage}
        onLongPress={() => {
          if (__DEV__) {
            selectionAsync()
            dispatch(openModal({ name: ModalName.Experiments }))
          }
        }}
        onPress={onPressAccountHeader}>
        {activeAddress && (
          <Flex row gap="xxs">
            <Flex shrink>
              <AddressDisplay
                hideAddressInSubtitle
                address={activeAddress}
                horizontalGap="xs"
                size={iconSizes.xl}
                variant="subheadLarge"
              />
            </Flex>
            <Chevron
              color={theme.colors.textSecondary}
              direction="s"
              height={iconSizes.lg}
              width={iconSizes.lg}
            />
          </Flex>
        )}
      </TouchableArea>
      <Flex alignItems="center" flexDirection="row" gap="md" justifyContent="flex-end">
        <QRScannerIconButton onPress={onPressScan} />
        <TouchableArea onPress={onPressNotifications} onPressIn={onPressInNotifications}>
          {sortedPendingTransactions?.length ? (
            <PendingNotificationBadge sortedPendingTransactions={sortedPendingTransactions} />
          ) : (
            <TxHistoryIconWithStatus />
          )}
        </TouchableArea>
      </Flex>
    </Box>
  )
}
