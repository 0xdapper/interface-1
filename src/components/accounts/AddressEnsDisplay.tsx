import React from 'react'
import { useAppTheme } from 'src/app/hooks'
import CopyIcon from 'src/assets/icons/copy-sheets.svg'
import { Button } from 'src/components/buttons/Button'
import { Flex } from 'src/components/layout'
import { Text } from 'src/components/Text'
import { ChainId } from 'src/constants/chains'
import { useENS } from 'src/features/ens/useENS'
import { ElementName } from 'src/features/telemetry/constants'
import { Theme } from 'src/styles/theme'
import { shortenAddress } from 'src/utils/addresses'
import { setClipboard } from 'src/utils/clipboard'

interface Props {
  address: Address | undefined
  mainSize?: number
  secondarySize?: number
  mainColor?: keyof Theme['colors']
  secondaryColor?: keyof Theme['colors']
  gap?: keyof Theme['spacing']
  showCopy?: boolean
  align?: 'center' | 'flex-start' | 'flex-end'
}

/**
 * Display dynamic stack of ENS and address, or only address if no ens
 */
export default function AddressEnsDisplay({
  address,
  mainSize = 16,
  secondarySize = 12,
  mainColor = 'deprecated_textColor',
  secondaryColor = 'deprecated_gray400',
  showCopy = true,
  gap = 'sm',
  align = 'flex-start',
}: Props) {
  const theme = useAppTheme()
  const ens = useENS(ChainId.Mainnet, address)

  const onPressCopyAddress = () => {
    if (!address) return
    setClipboard(address)
  }

  if (!address) {
    return null
  }

  return (
    <Flex alignItems={align} gap={gap}>
      {/* top line */}
      <Flex centered row gap="sm">
        <Text color={mainColor} fontSize={mainSize} fontWeight={'500'}>
          {ens.name ?? shortenAddress(address)}
        </Text>
        {showCopy && !ens.name && (
          <Button name={ElementName.Copy} onPress={onPressCopyAddress}>
            <CopyIcon color={theme.colors[mainColor]} height={mainSize} width={mainSize} />
          </Button>
        )}
      </Flex>
      {/* bottom line , only show if valid ens */}
      {ens.name && (
        <Flex centered row gap="sm">
          <Text color={secondaryColor} fontSize={secondarySize} fontWeight={'500'}>
            {shortenAddress(address)}
          </Text>
          {showCopy && (
            <Button name={ElementName.Copy} onPress={onPressCopyAddress}>
              <CopyIcon
                color={theme.colors[secondaryColor]}
                height={secondarySize}
                width={secondarySize}
              />
            </Button>
          )}
        </Flex>
      )}
    </Flex>
  )
}
