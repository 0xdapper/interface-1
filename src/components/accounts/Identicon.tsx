import { SpacingProps, SpacingShorthandProps } from '@shopify/restyle'
import React from 'react'
import { useColorScheme } from 'react-native'
import { useAppSelector } from 'src/app/hooks'
import { RemoteImage } from 'src/components/images/RemoteImage'
import { Box } from 'src/components/layout/Box'
import { selectUserLocalPfp } from 'src/features/wallet/walletSlice'
import { colorsDark, colorsLight } from 'src/styles/color'
import { theme, Theme } from 'src/styles/theme'
import { isValidAddress } from 'src/utils/addresses'

type Props = {
  address: Address
  size?: number
} & SpacingProps<Theme> &
  SpacingShorthandProps<Theme>

export function Identicon({ address, size = 36, ...rest }: Props) {
  if (!isValidAddress(address)) throw new Error(`Invalid address for identicon ${address}`)
  const isDarkMode = useColorScheme() === 'dark'

  const color = useAddressColor(address, isDarkMode)
  const userPfp = useAppSelector(selectUserLocalPfp)

  return (
    <Box
      borderRadius="full"
      height={size}
      style={{ backgroundColor: color }}
      width={size}
      {...rest}>
      {userPfp && (
        <RemoteImage
          borderRadius={theme.borderRadii.full}
          height={size}
          imageUrl={userPfp}
          width={size}
        />
      )}
    </Box>
  )
}

export function useAddressColor(address: string, isDarkMode: boolean, offset = 1) {
  const palette = isDarkMode ? colorsDark : colorsLight
  const colorSeed = parseInt(address.at(-offset)!, 16)
  if (colorSeed < 3) return palette.orange
  if (colorSeed < 6) return palette.green
  if (colorSeed < 9) return palette.pink
  if (colorSeed < 12) return palette.blue
  return palette.red
}
