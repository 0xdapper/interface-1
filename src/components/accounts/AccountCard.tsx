import React from 'react'
import TripleDots from 'src/assets/icons/triple-dots.svg'
import { Identicon } from 'src/components/accounts/Identicon'
import { Button } from 'src/components/buttons/Button'
import { CheckmarkCircle } from 'src/components/icons/CheckmarkCircle'
import { Box } from 'src/components/layout/Box'
import { CenterBox } from 'src/components/layout/CenterBox'
import { Text } from 'src/components/Text'
import { Account } from 'src/features/wallet/accounts/types'
import { shortenAddress } from 'src/utils/addresses'

interface Props {
  account: Account
  isActive?: boolean
  isEditable?: boolean
  onPress?: (address: Address) => void
  onEdit?: (address: Address) => void
}

export function AccountCard({
  account: { address, name },
  isActive,
  isEditable,
  onPress,
  onEdit,
}: Props) {
  return (
    <Button onPress={onPress ? () => onPress(address) : undefined} width="100%">
      <Box
        width="100%"
        flexDirection="row"
        alignItems="center"
        justifyContent="space-between"
        backgroundColor="white"
        testID={`account_item/${address.toLowerCase()}`}>
        <CenterBox flexDirection="row">
          <Identicon address={address} size={50} mr="md" />
          <Box>
            <Text variant="body" fontWeight="500">
              {name || shortenAddress(address)}
            </Text>
            {/* TODO get real total */}
            <Text variant="bodySm" color="gray400" mt="xs">
              {shortenAddress(address)}
            </Text>
          </Box>
        </CenterBox>
        {isEditable && onEdit && (
          <Button onPress={() => onEdit(address)} mx="sm" my="md">
            <TripleDots width={22} height={12} />
          </Button>
        )}
        {!isEditable && isActive && <CheckmarkCircle size={30} backgroundColor="green" />}
      </Box>
    </Button>
  )
}
