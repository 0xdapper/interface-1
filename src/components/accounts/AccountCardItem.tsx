import React from 'react'
import { useColorScheme } from 'react-native'
import { useAppTheme } from 'src/app/hooks'
import QrCode from 'src/assets/icons/qr-code.svg'
import TripleDots from 'src/assets/icons/triple-dots.svg'
import { useAddressColor } from 'src/components/accounts/Identicon'
import { AddressDisplay } from 'src/components/AddressDisplay'
import { Button } from 'src/components/buttons/Button'
import { Flex } from 'src/components/layout'
import { TotalBalance } from 'src/features/balances/TotalBalance'
import { useActiveChainIds } from 'src/features/chains/utils'
import { useAllBalancesByChainId } from 'src/features/dataApi/balances'
import { ElementName } from 'src/features/telemetry/constants'
import { Account } from 'src/features/wallet/accounts/types'
import { opacify } from 'src/utils/colors'

interface Props {
  account: Account
  isActive?: boolean
  onPress?: (address: Address) => void
  onPressQRCode: (address: Address) => void
  onPressEdit?: (address: Address) => void
}

export function AccountCardItem({ account, isActive, onPress, onPressQRCode, onPressEdit }: Props) {
  const { address } = account
  const isDarkMode = useColorScheme() === 'dark'
  const theme = useAppTheme()

  const color = useAddressColor(address, isDarkMode)

  const currentChains = useActiveChainIds()
  const { balances } = useAllBalancesByChainId(currentChains)

  return (
    <Button onPress={onPress ? () => onPress(address) : undefined}>
      <Flex
        borderRadius="lg"
        borderWidth={isActive ? 1.5 : 1}
        flexDirection="column"
        gap="xl"
        my="xs"
        p="md"
        style={{
          borderColor: isActive ? theme.colors.primary1 : opacify(80, color),
          backgroundColor: opacify(20, color),
        }}
        testID={`account_item/${address.toLowerCase()}`}>
        <Flex row alignItems="center" borderRadius="sm" justifyContent="space-between">
          <AddressDisplay
            alwaysShowAddress
            address={address}
            size={40}
            variant="body"
            verticalGap="none"
          />
          <Button
            alignItems="center"
            height={20}
            justifyContent="center"
            padding="sm"
            width={20}
            onPress={() => onPressQRCode(address)}>
            <QrCode height={20} stroke={theme.colors.textColor} width={20} />
          </Button>
        </Flex>
        <Flex row alignItems="center" justifyContent="space-between">
          <TotalBalance balances={balances} variant="h3" />
          {onPressEdit && (
            <Button name={ElementName.Edit} onPress={() => onPressEdit(address)}>
              <TripleDots
                height={12}
                stroke={isDarkMode ? opacify(50, '#FFFFFF') : opacify(30, '#000000')}
                strokeLinecap="round"
                strokeWidth="2"
                width={20}
              />
            </Button>
          )}
        </Flex>
      </Flex>
    </Button>
  )
}
