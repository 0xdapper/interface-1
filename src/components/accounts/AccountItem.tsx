import React from 'react'
import TripleDots from 'src/assets/icons/triple-dots.svg'
import { Identicon } from 'src/components/accounts/Identicon'
import { Button } from 'src/components/buttons/Button'
import { CheckmarkCircle } from 'src/components/icons/CheckmarkCircle'
import { Box } from 'src/components/layout/Box'
import { CenterBox } from 'src/components/layout/CenterBox'
import { Text } from 'src/components/Text'
import { Account } from 'src/features/wallet/accounts/types'
import { shortenAddress } from 'src/utils/addresses'

interface Props {
  account: Account
  isActive?: boolean
  isEditable?: boolean
  onPress?: (address: Address) => void
  onEdit?: (address: Address) => void
}

export function AccountItem({
  account: { address, name },
  isActive,
  isEditable,
  onPress,
  onEdit,
}: Props) {
  return (
    <Button width="100%" onPress={onPress ? () => onPress(address) : undefined}>
      <Box
        alignItems="center"
        backgroundColor="white"
        flexDirection="row"
        justifyContent="space-between"
        testID={`account_item/${address.toLowerCase()}`}
        width="100%">
        <CenterBox flexDirection="row">
          <Identicon address={address} mr="md" size={50} />
          <Box>
            <Text fontWeight="500" variant="body">
              {name || shortenAddress(address)}
            </Text>
            {/* TODO get real total */}
            <Text color="gray400" mt="xs" variant="bodySm">
              {shortenAddress(address)}
            </Text>
          </Box>
        </CenterBox>
        {isEditable && onEdit && (
          <Button mx="sm" my="md" onPress={() => onEdit(address)}>
            <TripleDots height={12} width={22} />
          </Button>
        )}
        {!isEditable && isActive && <CheckmarkCircle backgroundColor="green" size={30} />}
      </Box>
    </Button>
  )
}
