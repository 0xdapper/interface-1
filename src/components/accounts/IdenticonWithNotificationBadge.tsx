import React from 'react'
import { Identicon } from 'src/components/accounts/Identicon'
import { NotificationBadge } from 'src/components/notifications/Badge'
import { useSelectAddressNotificationCount } from 'src/features/notifications/hooks'

interface Props {
  address: string
  size: number
}

export function IdenticonWithNotificationBadge({ address, size }: Props) {
  const notificationCount = useSelectAddressNotificationCount(address)

  if (!notificationCount) {
    return <Identicon address={address} size={size} />
  }

  return (
    <NotificationBadge backgroundColor="accentAction" notificationCount={notificationCount}>
      <Identicon address={address} size={size} />
    </NotificationBadge>
  )
}
