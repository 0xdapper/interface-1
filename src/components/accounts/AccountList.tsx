import { LinearGradient } from 'expo-linear-gradient'
import { ComponentProps, default as React, useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import { FlatList, ListRenderItemInfo, StyleSheet } from 'react-native'
import { useAppTheme } from 'src/app/hooks'
import PlusIcon from 'src/assets/icons/plus.svg'
import { AccountCardItem } from 'src/components/accounts/AccountCardItem'
import { TouchableArea } from 'src/components/buttons/TouchableArea'
import { Box, Flex } from 'src/components/layout'
import { Text } from 'src/components/Text'
import { Account, AccountType } from 'src/features/wallet/accounts/types'
import { useActiveAccount } from 'src/features/wallet/hooks'
import { spacing } from 'src/styles/sizing'

type AccountListProps = Pick<ComponentProps<typeof AccountCardItem>, 'onPress' | 'onPressEdit'> & {
  accounts: Account[]
  onAddWallet: () => void
}

const STICKY_HEADER_INDICES = [0]

export function AccountList({ accounts, onAddWallet, onPressEdit, onPress }: AccountListProps) {
  const { t } = useTranslation()
  const theme = useAppTheme()

  const activeAccount = useActiveAccount()

  const ListHeader = useMemo(
    () => (
      <Flex row alignItems="center" bg="background0" borderBottomColor="backgroundOutline" pb="sm">
        <Box flex={1}>
          <Text color="textPrimary" px="lg" variant="bodyLarge">
            {t('Your wallets')}
          </Text>
        </Box>
        <TouchableArea
          borderColor="backgroundOutline"
          borderRadius="full"
          borderWidth={1}
          mr="md"
          p="xs"
          onPress={onAddWallet}>
          <PlusIcon
            color={theme.colors.textSecondary}
            height={theme.iconSizes.sm}
            width={theme.iconSizes.sm}
          />
        </TouchableArea>
      </Flex>
    ),
    [t, theme, onAddWallet]
  )

  const renderItem = useMemo(
    () =>
      ({ item }: ListRenderItemInfo<Account>) => {
        return (
          <AccountCardItem
            account={item}
            isActive={!!activeAccount && activeAccount.address === item.address}
            isViewOnly={item.type === AccountType.Readonly}
            onPress={onPress}
            onPressEdit={onPressEdit}
          />
        )
      },
    [activeAccount, onPress, onPressEdit]
  )

  return (
    <Box flexShrink={1} position="relative">
      <FlatList
        ListFooterComponent={<Box height={theme.spacing.lg} />}
        ListHeaderComponent={ListHeader}
        bounces={false}
        data={accounts}
        keyExtractor={key}
        renderItem={renderItem}
        showsVerticalScrollIndicator={false}
        stickyHeaderIndices={STICKY_HEADER_INDICES}
      />
      <LinearGradient
        colors={[theme.colors.translucentBackgroundBackdrop, theme.colors.background0]}
        end={{ x: 0, y: 1 }}
        start={{ x: 0, y: 0 }}
        style={ListSheet.gradient}
      />
    </Box>
  )
}

const key = (account: Account) => account.address

const ListSheet = StyleSheet.create({
  gradient: {
    bottom: 0,
    height: spacing.lg,
    left: 0,
    position: 'absolute',
    width: '100%',
  },
})
