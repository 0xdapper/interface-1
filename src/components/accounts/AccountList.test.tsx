import { MockedResponse } from '@apollo/client/testing'
import React from 'react'
import { AccountList } from 'src/components/accounts/AccountList'
import { AccountListDocument, AccountListQuery } from 'src/data/__generated__/types-and-hooks'
import { account } from 'src/test/fixtures'
import { Portfolios } from 'src/test/gqlFixtures'
import { render, screen } from 'src/test/test-utils'
import { formatUSDPrice, NumberType } from 'src/utils/format'

jest.mock('@react-navigation/drawer', () => ({ useDrawerStatus: jest.fn(() => 'unknown') }))

const mock: MockedResponse<AccountListQuery> = {
  request: {
    query: AccountListDocument,
    variables: {
      addresses: [account.address],
    },
  },
  result: {
    data: {
      portfolios: Portfolios,
    },
  },
}

describe(AccountList, () => {
  it('renders without error', async () => {
    const tree = render(
      <AccountList
        accounts={[account]}
        onAddWallet={function (): void {
          throw new Error('Function not implemented.')
        }}
      />,
      { mocks: [mock] }
    )

    expect(
      await screen.findByText(
        formatUSDPrice(
          Portfolios[0].tokensTotalDenominatedValue?.value,
          NumberType.PortfolioBalance
        )
      )
    ).toBeDefined()
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
