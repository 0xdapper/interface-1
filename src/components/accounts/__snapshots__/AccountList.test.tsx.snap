// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountList renders without error 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexShrink": 1,
        "position": "relative",
      },
    ]
  }
>
  <RCTScrollView
    ListFooterComponent={
      <ForwardRef
        height={24}
      />
    }
    ListHeaderComponent={
      <Flex
        alignItems="center"
        bg="background0"
        borderBottomColor="backgroundOutline"
        pb="sm"
        row={true}
      >
        <ForwardRef
          flex={1}
        >
          <Text
            color="textPrimary"
            px="lg"
            variant="bodyLarge"
          >
            Your wallets
          </Text>
        </ForwardRef>
        <TouchableArea
          borderColor="backgroundOutline"
          borderRadius="full"
          borderWidth={1}
          mr="md"
          onPress={[Function]}
          p="xs"
        >
          <SvgPlus
            color="#505A78"
            height={12}
            width={12}
          />
        </TouchableArea>
      </Flex>
    }
    bounces={false}
    data={
      Array [
        Object {
          "account": Object {
            "address": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
            "derivationIndex": 0,
            "mnemonicId": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
            "name": "Test Account",
            "timeImportedMs": 10,
            "type": "signerMnemonic",
          },
          "isPortfolioValueLoading": false,
          "portfolioValue": 55,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderBottomColor": "#D2D9EE",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingBottom": 12,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "flex": 1,
                },
              ]
            }
          >
            <Text
              allowFontScaling={true}
              maxFontSizeMultiplier={1.4}
              style={
                Array [
                  Object {
                    "color": "#0D111C",
                    "display": "flex",
                    "fontFamily": "Inter-Medium",
                    "fontSize": 16,
                    "lineHeight": 24,
                    "paddingHorizontal": 24,
                  },
                ]
              }
            >
              Your wallets
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "height": 16,
                  "width": 16,
                },
              ]
            }
          />
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 5,
                "left": 5,
                "right": 5,
                "top": 5,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderColor": "#D2D9EE",
                "borderRadius": 999999,
                "borderWidth": 1,
                "display": "flex",
                "marginRight": 16,
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            <svg
              color="#505A78"
              data-file-name="SvgPlus"
              height={12}
              width={12}
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 5,
              "left": 5,
              "right": 5,
              "top": 5,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "display": "flex",
              "opacity": 1,
              "paddingBottom": 12,
              "paddingHorizontal": 24,
              "paddingTop": 8,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                },
              ]
            }
            testID="account_item/0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "flexShrink": 1,
                    "justifyContent": "flex-start",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "display": "flex",
                      "position": "relative",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderColor": "transparent",
                        "borderRadius": 999999,
                        "borderWidth": 0,
                        "display": "flex",
                        "position": "relative",
                      },
                      Object {
                        "padding": 0,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "display": "flex",
                          "height": 28,
                          "width": 28,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        clip={true}
                        mask={
                          <UniconMask
                            attributeData={
                              Object {
                                "0": "#5CFE9D",
                                "1": "#B8C3B7",
                                "2": Array [
                                  Object {
                                    "path": "M34 20C34 28.8366 26.8366 36 18 36C9.16344 36 2 28.8366 2 20L2 -3.8147e-06L34 -1.01717e-06V20Z",
                                  },
                                ],
                                "3": Array [
                                  Object {
                                    "fillType": "evenOdd",
                                    "path": "M16 8C12.2937 6.62854 9.37146 3.70632 8 0C6.62854 3.70632 3.70632 6.62854 0 8C3.70632 9.37146 6.62854 12.2937 8 16C9.37146 12.2937 12.2937 9.37146 16 8ZM8 11C9.65685 11 11 9.65685 11 8C11 6.34315 9.65685 5 8 5C6.34315 5 5 6.34315 5 8C5 9.65685 6.34315 11 8 11Z",
                                  },
                                ],
                              }
                            }
                            size={28}
                          />
                        }
                      >
                        <View
                          transform={
                            Array [
                              Object {
                                "scale": 0.7567567567567568,
                              },
                            ]
                          }
                        >
                          <View
                            height={37}
                            width={37}
                            x={0}
                            y={0}
                          >
                            <View
                              colors={
                                Array [
                                  "#5CFE9D",
                                  "#B8C3B7",
                                ]
                              }
                              end={null}
                              start={null}
                            />
                          </View>
                          <View
                            color="#EDE590"
                            cx={18.5}
                            cy={-13.36111111111111}
                            r={30.833333333333336}
                          >
                            <View
                              blur={15}
                            />
                          </View>
                        </View>
                      </View>
                      <View
                        clip={true}
                        mask={
                          <UniconMask
                            attributeData={
                              Object {
                                "0": "#5CFE9D",
                                "1": "#B8C3B7",
                                "2": Array [
                                  Object {
                                    "path": "M34 20C34 28.8366 26.8366 36 18 36C9.16344 36 2 28.8366 2 20L2 -3.8147e-06L34 -1.01717e-06V20Z",
                                  },
                                ],
                                "3": Array [
                                  Object {
                                    "fillType": "evenOdd",
                                    "path": "M16 8C12.2937 6.62854 9.37146 3.70632 8 0C6.62854 3.70632 3.70632 6.62854 0 8C3.70632 9.37146 6.62854 12.2937 8 16C9.37146 12.2937 12.2937 9.37146 16 8ZM8 11C9.65685 11 11 9.65685 11 8C11 6.34315 9.65685 5 8 5C6.34315 5 5 6.34315 5 8C5 9.65685 6.34315 11 8 11Z",
                                  },
                                ],
                              }
                            }
                            overlay={true}
                            size={28}
                          />
                        }
                      >
                        <View
                          color="#000000"
                          height={37}
                          opacity={0.08}
                          width={37}
                          x={0}
                          y={0}
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "display": "flex",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "height": 16,
                      "width": 16,
                    },
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "stretch",
                      "display": "flex",
                      "flex": 1,
                      "flexDirection": "column",
                      "justifyContent": "flex-start",
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={true}
                  maxFontSizeMultiplier={1.4}
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#0D111C",
                        "display": "flex",
                        "fontFamily": "Inter-Medium",
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                    ]
                  }
                >
                  0x​f39F...2266
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "display": "flex",
                        "flexGrow": 0,
                        "flexShrink": 0,
                        "height": 0,
                      },
                    ]
                  }
                />
                <Text
                  allowFontScaling={true}
                  maxFontSizeMultiplier={1.4}
                  style={
                    Array [
                      Object {
                        "color": "#505A78",
                        "display": "flex",
                        "fontFamily": "Inter-Regular",
                        "fontSize": 14,
                        "lineHeight": 20,
                      },
                    ]
                  }
                >
                  $55
                  <Text
                    allowFontScaling={true}
                    maxFontSizeMultiplier={1.4}
                    style={
                      Array [
                        Object {
                          "color": "#888FAB",
                          "display": "flex",
                          "fontFamily": "Inter-Regular",
                          "fontSize": 14,
                          "lineHeight": 20,
                        },
                      ]
                    }
                  >
                    .
                    00
                  </Text>
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "height": 16,
                    "width": 16,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "height": 24,
              },
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
  ExpoLinearGradient
</View>
`;
